### ISSUE TITLE
How to use with start-server-and-test?

### ISSUE THREAD

**Issue by anishkny** [stripe/stripe-cli#760]
_Copying over from https://github.com/stripe/stripe-cli/discussions/756 for visibility._

I am trying to use this in conjunction with [start-server-and-test](https://github.com/bahmutov/start-server-and-test) which starts a server, waits for a port/URL to be available and then runs a test. What port/URL should I instruct it to listen on to ensure the stripe CLI is online?

FYI, I am using the Docker image flavor (not the native CLI).

TIA

**Comment by pepin-stripe** [comment]
Hi @anishkny, sorry I'm sure I understand what you are asking - I'm not familiar with start-server-and-test either so unsure of the use case here. The CLI doesn't start a server, and doesn't allocate any port/url.

Can you tell us a bit more what you are trying to do?

**Comment by anishkny** [comment]
@pepin-stripe thanks for getting back.

I am trying to write integration tests for a Next.js + Stripe project [anishkny/stripe-nextjs-test](https://github.com/anishkny/stripe-nextjs-test). In it, I want to start `stripe listen`, wait for it to be ready, and then start my test.

`start-server-and-test` is a utility that lets you start a server, wait for a TCP port to be used by that server and then run your test (more help at [repo](https://github.com/bahmutov/start-server-and-test))

For example:

```
npx start-server-and-test start 8080 test
```

will run `npm start`, wait for port 8080 to be used and then run `npm test`. Finally, it also terminates the server so that one can use this in CI.

So basically how do I determine whether `stripe listen` is ready for testing in an automated manner.

Hope this made some more sense 😄

**Comment by pepin-stripe** [comment]
Ok I think I understand a bit more.

The CLI does not start a server, so `start-server-and-test` will not be able to help on starting `stripe listen`, monitoring when it's ready and killing it when it's done.

Instead, you can use node's [child_process](https://nodejs.org/api/child_process.html#child_process_child_process_spawn_command_args_options) package to spawn a `stripe listen` process, and monitoring for the `[spawn](https://nodejs.org/api/child_process.html#child_process_event_spawn)` event to confirm that the process started successfully and is ready:

Something like this:
```js
const {spawn} = require('child_process')

const listenProc = spawn('stripe', 'listen')
listenProc.on('spawn', () => console.log('stripe listen ready!');

// later on once you're done
listenProc.kill('SIGINT');
```

We don't have a great way to do integration tests with webhook events or use the CLI for it (we're actively looking at possible solution).

I'm gonna close this ticket now, but feel free to re-open if you have additional questions.

Cheers!
